### PR TITLE
UI: pki auto-tidy views

### PR DESCRIPTION
**Pull request by hellobontempo**
Adds auto-tidy functionality to the UI!

Part of PKI Phase 2 UI work: https://github.com/hashicorp/vault/pull/20706

### CODE REVIEW

File: ui/app/models/pki/tidy.js [hunk 1]
```diff
@@ -4,9 +4,167 @@
...
-  @attr('boolean', { defaultValue: false }) tidyRevocationQueue;
-  @attr('string', { defaultValue: '72h' }) safetyBuffer;
+  // the backend mount is the model id, only one pki/tidy model will ever persist (the auto-tidy config)
+  @service version;
+
+  @attr({
+    label: 'Tidy ACME enabled',
+    labelDisabled: 'Tidy ACME disabled',
+    mapToBoolean: 'tidyAcme',
+    helperTextDisabled: 'Tidying of ACME accounts,  orders and authorizations is disabled',
```
Comment (hellobontempo) [inline ui/app/models/pki/tidy.js:19]:
```suggestion
    helperTextDisabled: 'Tidying of ACME accounts, orders and authorizations is disabled',
```

File: ui/app/models/pki/tidy.js [hunk 2]
```diff
@@ -4,9 +4,167 @@
...
+    subText: 'Remove all invalid and expired certificates from storage.',
+  })
+  tidyRevokedCerts;
+
+  get useOpenAPI() {
+    return true;
+  }
+  getHelpUrl(backend) {
+    return `/v1/${backend}/config/auto-tidy?help=1`;
+  }
```
Comment (hellobontempo) [inline ui/app/models/pki/tidy.js:129]:
```suggestion
  get useOpenAPI() {
    return true;
  }
  
  getHelpUrl(backend) {
    return `/v1/${backend}/config/auto-tidy?help=1`;
  }
```

File: ui/lib/pki/README.md [hunk 3]
```diff
@@ -32,6 +32,18 @@ If you couldn't tell from the documentation above, PKI is _complex_. As such, th
 
   The `parsedCertificate` attribute is an object that houses all of the parsed certificate data returned by the [parse-pki-cert.js](../../app/utils/parse-pki-cert.js) util.
 
+- ### [pki/tidy](../../app/models/pki/tidy.js)
+
+  This model is used to manage [tidy](https://developer.hashicorp.com/vault/api-docs/secret/pki#tidy) operations in a few different contexts. All of the following endpoints share the same parameters _except_ `enabled` and `interval_duration` which are reserved for auto-tidy configuration settings.
```
Comment (hellobontempo) [inline ui/lib/pki/README.md:37]:
```suggestion
  This model is used to manage [tidy](https://developer.hashicorp.com/vault/api-docs/secret/pki#tidy) operations in a few different contexts. All of the following endpoints share the same parameters _except_ `enabled` and `interval_duration` which are reserved for auto-tidy operations only.
```

File: ui/lib/pki/addon/components/page/pki-tidy-status.hbs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      <ol class="has-left-margin-m has-bottom-margin-s">
+        <li>Select a tidy operation:</li>
+        <ul class="bullet has-bottom-margin-xs">
+          <li><strong>Automatic tidy</strong>
+            periodically runs a tidy operation with saved configuration settings after waiting the specified interval
+            duration between tidies
+          </li>
+          <li><strong>Manual tidy</strong> runs a tidy operation once</li>
+        </ul>
+        <li>Configure the parameters that determine how to tidy, and run the operation</li>
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/page/pki-tidy-status.hbs:130]:
```suggestion
        <li>Configure the parameters that determine how to tidy and run the operation.</li>
```

File: ui/lib/pki/addon/routes/tidy/manual.js [hunk 5]
```diff
@@ -5,10 +5,10 @@
 
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
-import { withConfirmLeave } from 'core/decorators/confirm-leave';
+// import { withConfirmLeave } from 'core/decorators/confirm-leave';
```
Comment (hellobontempo) [inline ui/lib/pki/addon/routes/tidy/manual.js:8]:
```suggestion
import { withConfirmLeave } from 'core/decorators/confirm-leave';
```

File: ui/lib/pki/addon/routes/tidy/manual.js [hunk 6]
```diff
@@ -5,10 +5,10 @@
...
 
-@withConfirmLeave('model.tidy')
-export default class PkiConfigurationTidyRoute extends Route {
+// @withConfirmLeave()
```
Comment (hellobontempo) [inline ui/lib/pki/addon/routes/tidy/manual.js:10]:
```suggestion
@withConfirmLeave()
```

File: ui/lib/pki/addon/controllers/tidy/index.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+const POLL_INTERVAL_MS = 5000;
+
+export default class PkiTidyIndexController extends Controller {
+  @service store;
+  @service secretMountPath;
+
+  @tracked tidyStatus = null;
+
+  @task
+  *pollTidyStatus() {
```
Comment (hashishaw) [inline ui/lib/pki/addon/controllers/tidy/index.js:20]: We should probably cancel when the user moves away from Tidy -- in the cluster route we do:
```
poll: task(function* () {
    while (true) {
      // do stuff
    }
  })
    .cancelOn('deactivate')
```
Not sure off the top of my head how that translates to class-based

File: ui/tests/acceptance/pki/pki-tidy-test.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    await click(SELECTORS.tidyConfigureModal.tidyModalManualButton);
+    assert.dom(SELECTORS.tidyForm.tidyFormName('manual')).exists('Manual tidy form exists');
+    await click(SELECTORS.tidyForm.inputByAttr('tidyCertStore'));
+    await fillIn(SELECTORS.tidyForm.tidyPauseDuration, '10');
+    await click(SELECTORS.tidyForm.tidySave);
+    await click(SELECTORS.cancelTidyAction);
+    assert.dom(SELECTORS.cancelTidyModalBackground).exists('Confirm cancel tidy modal exits');
+    await click(SELECTORS.tidyConfigureModal.tidyModalCancelButton);
+    // we can't properly test the background refresh fetching of tidy status in testing
+    this.server.get(`${this.mountPath}/tidy-status`, () => {
```
Comment (hashishaw) [inline ui/tests/acceptance/pki/pki-tidy-test.js:61]: Should this be set on the beforeEach hook so that it's always available to future tests as well?

**Review by hashishaw — CHANGES_REQUESTED** [review]
Great work, and love the test coverage! 🤩 I think the only blocker is canceling the Tidy poll when the user moves away from the Tidy area.

File: ui/tests/acceptance/pki/pki-tidy-test.js (hunk 8 above)
Comment (kiannaquach) [reply ui/tests/acceptance/pki/pki-tidy-test.js:61]: @hellobontempo added an empty state for tidy and in order to configure tidy the acceptance tests go through the empty states and configure based on the test criteria.

**Review by hashishaw — APPROVED** [review]
Simple and beautiful ✨

File: ui/tests/acceptance/pki/pki-tidy-test.js (hunk 8 above)
Comment (hellobontempo) [reply ui/tests/acceptance/pki/pki-tidy-test.js:61]: We skip the polling loop in testing as well, so this is the mocked response after clicking `cancel tidy`

File: ui/lib/pki/addon/controllers/tidy/index.js (hunk 7 above)
Comment (kiannaquach) [reply ui/lib/pki/addon/controllers/tidy/index.js:21]: good catch thanks to claire for finding that `resetController` has a `isExiting` arg :)

**Pull request merged**